File: javascript/Insertion_sort.js
// insertion sort algorithm

var array = [34, 12, 24, 9, 5];

function insertionSort(array) {
    for (var i = 0; i < array.length; i++) {
        var temp = array[i];
        for (var j = i - 1; j >= 0 && array[j] > temp; j--) {
            array[j + 1] = array[j];
        }
        array[j + 1] = temp;
    }
    return array;
}

console.log(insertionSort(array));

File: CPP/Sorting/insertionSort.cpp
#include <iostream>
using namespace std;
int main() {
    // Insertion Sort
    int n,j;
    cin>>n;
    int a[n];
    for(int i=0;i<n;i++){cin>>a[i];}
    for(int i=0;i<n;i++){
        int temp=a[i];
        for(j=i-1;j>=0 && a[j]>temp;j--){a[j+1]=a[j];}a[j+1]=temp;}
    for(int i=0;i<n;i++){cout<<a[i]<<" ";}
}
